### ISSUE TITLE
Some bugs/typos in Fed 2 composition hints

### ISSUE THREAD

**Issue by lennyburdette** [apollographql/federation#1659]
1. `InconsistentObjectValueTypeField`:
     The `elementCoordinate` is the type name, while the `mainElementDescription` indicates that it should be the field:
     
     > Product: the field that is inconsistently declared between subgraphs

2. `InconsistentArgumentPresence`:
      Looks like some values are missing: 

      > Argument "Inconsistent.missingArg(a:)" will not be added to "missingArg(a: undefined): undefined" in the supergraph as it does not appear in all subgraphs: it is defined in subgraph "a" but not in subgraph "b".

3. Nitpick: "executable directive" instead of "execution directive" — the spec called them "ExecutableDirectiveLocations".
4. `NoExecutionDirectiveLocationsIntersection`:
     > Execution directive "@bar" has no location that is common to all subgraphs: it will not appear in the **~subgraph~ supergraph** as there no intersection between location "FIELD" in subgraphs "a" and "c".
5. `InconsistentEnumValue`:
      > MyEnum: the **~union~ enum** type which has an inconsistent member
6. I was able to get redundant `InconsistentUnionMember` and `InconsistentEnumValue` hints with three subgraphs. This hint showed up twice:
      > Member type "Inconsistent" in union type "MyUnion" is only defined in a subset of subgraphs defining "MyUnion" (but can always be resolved from these subgraphs): "Inconsistent" is defined in subgraph "b" but not in subgraphs "a" and "c".
7. I couldn’t get *InconsistentExecutionDirectiveLocations* to appear. I thought that this would do it:
      ```
      directive @baz on FIELD | OBJECT
      # and
      directive @baz on FIELD | INTERFACE
      ```

**Comment by pcmanus** [comment]
I've pushed #1683 which does a bunch of changes to hints and should all the points here.  But to expand on a few of those points more directly:

> The `elementCoordinate` is the type name

I've removed the `elementCoordinate` part of hints for now. The rational being that we want hints to be behave essentially similarly to errors (so they can be handled consistently), and errors don't have that. I don't exclude adding this back at some point, but the truth is that this part hadn't been though thru too much so I'm reluctant to commit to it for GA before we give a bit more thought.

> I couldn’t get InconsistentExecutionDirectiveLocations to appear.

It only take executable locations into account, so in the example, only `FIELD` is taken into account and it isn't inconsistent. You'd have to use, say `FIELD | INLINE_FRAGMENT` in one subgraph and just `FIELD` in another.

As an aside, there is currently no hint for the fact that type-system level locations are just ignored (the underlying reason, which is not new to fed2 but inherited from fed1, is that executable directives are merged, but type-system level ones aren't, at least by default), but we probably should add one eventually. It would just be a different one.
